lib/gp-iam.js

```diff
@@ -29,12 +42,59 @@ const GpIAM = function GpIAM(credentials) {
   if(!this.credentials || !this.credentials.apikey || !this.credentials.iam_endpoint) {
```

Comment:
Since you are allowing both api key and token for iam auth, you might want to keep check for iam endpoint conditional.

Reply:
@sid41 the API states that the `iam_endpoint` is required, I wanted to keep the credential requirement consistent. Using the `API-KEY` header is not intended to be a high runner use case.